### DISCUSSION TITLE
Support for Azure Managed Identity SQL Server Login

### DISCUSSION THREAD

**Discussion by aluchs-hpg** [directus/directus#13687] [Feature Requests]
In order to improve security posture and development agility, it would be great if directus could support connecting to MS SQL Server via Azure Managed Identity instead of username / password. This allows for connectivity without secret management and potential risk of password compromise.

[Simple example documentation from Microsoft.](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/tutorial-windows-vm-access-sql)

**Comment by rijkvanzanten** [comment]
That was actually just implemented in https://github.com/directus/directus/pull/11141 which will be released shortly 🙂

**Reply by aluchs-hpg** [reply]
Oh wow! I'm sorry, I only searched issues and discussions for "managed identity" I didn't search PRs. My apologies for the clutter, and thank you so much for the implementation and fast response.

**Reply by aluchs-hpg** [reply]
Although... looking through the pull request, I might be missing osmething but it looks like for [MS SQL Server, DB_USER and DB_PASSWORD are still required](https://github.com/directus/directus/pull/11141/files#diff-9516a9253e5c379aaf617b5197eeaaa5f3cd08b05cd5fd65655fc670c54735e8R54)?

EDIT: Agh, link doesn't work because line in question is hidden by default but its pointing to line 56 in api/src/database/index.ts on the right side of the diff.

**Reply by rijkvanzanten** [reply]
Already taken care of https://github.com/directus/directus/pull/13518 🙂

**Reply by aluchs-hpg** [reply]
Oh okay, I think I didn't understand the purpose/semantics of DB_TYPE, but that will be non default for this use case. Got it. Thanks again!

**Reply by rijkvanzanten** [reply]
The `type` flag is how `tedious` (MS SQL connection lib) handles the different auth types; Directus simply passes every `DB_*` env var as a serialized object to the connection library, so therefore it becomes `DB_TYPE` in Directus 🙂
